wireapp/kalium
# refactor: remove self_deletion_start_date [WPB-6219]

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

`self_deletion_start_date` is deprecated and should be removed and fully replaced by `self_deletion_end_date` but it is still kept in the database and used in some places.

### Solutions

Use `ALTER TABLE DROP COLUMN` which is available from SQLite 3.35.0 - we provide our own sql binaries and for prod we use 3.44 (sqlcipher), it's needed to also update sqldelight dialect.
Note: this migration will not work on debug builds where we don't use sqlcipher to allow to inspect the db for debugging purposes.

### Testing

#### Test Coverage (Optional)


#### How to Test

Send some self-deleting messages, update the app to perform DB migration and check if everything is ok with these messages.

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
